src/utilities/theme/role-variants.ts

```diff
@@ -136,43 +136,43 @@ export const roleVariants: RoleVariants = {
       description:
         'For use as a subdued border (border or interactive outline).',
       light: {lightness: 85},
-      dark: {lightness: 15},
+      dark: {lightness: 32},
     },
     {
       name: 'iconOnSurface',
       description: 'For use as the fill color of neutral icons.',
-      light: {lightness: 29.9},
-      dark: {lightness: 98},
+      light: {lightness: 40.1},
+      dark: {lightness: 70.1},
     },
     {
       name: 'iconDisabledOnSurface',
       description: 'For use as the fill color of disabled neutral icons.',
-      light: {lightness: 68},
-      dark: {lightness: 75},
+      light: {lightness: 76.9},
+      dark: {lightness: 36.8},
     },
     {
       name: 'iconSubduedOnSurface',
       description: 'For use as the fill color of subdued neutral icons.',
       light: {lightness: 59.8},
-      dark: {lightness: 62},
+      dark: {lightness: 52.1},
     },
     {
       name: 'textOnSurface',
       description: 'For use as a neutral text color.',
-      light: {lightness: 13},
-      dark: {lightness: 100},
+      light: {lightness: 13.1},
+      dark: {lightness: 90.8},
```

Comment:
Per my comment about accessibility, this value will need to go lighter to get `textOnSurfaceDark` to pass color contrast.

Reply:
It _shouldn't_. Could this be pointing to surface contrast issue somewhere else?

Reply:
Thinking about this a bit more, and I think the problem is more to do with the fact that we should be using textOnPrimary for primary action text, and not textOnSurfaceDark. Likewise, we should create a textOnCritical variant for use as the critical action text, and not use textOnSurfaceDark for that either.